docs/okeps/okep-5193-user-defined-networks.md

```diff
@@ -0,0 +1,1468 @@
+and will be addressed in a future enhancement. DNS lookups for services and external entities will function correctly.
+
+### Services
+
+Services in Kubernetes are namespace scoped. Any creation of a service in a namespace without a user-defined network
+(using cluster default network as primary) will only be accessible by other namespaces also using the default network as
+their primary network. Services created in namespaces served by user-defined networks, will only be accessible to
+namespaces connected to the user-defined network.
+
+Since most applications require DNS and KAPI access, there is an exception to the above conditions where pods that are
+connected to user-defined networks are still able to access KAPI and DNS services that reside on the cluster default
+network. In the future, access to more services on the default network may be granted. However, that would require more
+groundwork around enforcing network policy (which is evaluated typically after service DNAT) as potentially nftables
+rules. Such work is considered a future enhancement and beyond the scope of this initial implementation.
+
+With this proposal, OVN-Kubernetes will check which network is being used for this namespace, and then only enable the
+service there. The cluster IP of the service will only be available in the network of that service, except for KAPI and
+DNS as previously explained. Host networked pods in a namespace with a user-defined primary network will also be limited
+to only accessing the cluster IP of the services for that network. Load balancer IP and nodeport services are also
+supported on user-defined networks. Service selectors are only able to select endpoints from the same namespace where the
+service exists. Services that exist before the user-defined network is assigned to a namespace will result in
+OVN-Kubernetes executing a re-sync on all services in that namespace, and updating all load balancers. Keep in mind that
+pods must not exist in the namespace when the namespace is assigned to a new network or the new network assignment will
+not be accepted by OVN-Kubernetes.
+
+Services in a user-defined network will be reachable by other namespaces that share the same network.
+
+As previously mentioned, Kubernetes API and DNS services will be accessible by all pods.
+
+Endpoint slices will provide the IPs of the cluster default network in Kubernetes API. For this implementation the required
+endpoints are those IP addresses which reside on the user-defined primary network. In order to solve this problem,
+OVN-Kubernetes may create its own endpoint slices or may choose to do dynamic lookups at runtime to map endpoints to
+their primary IP address. Leveraging a second set of endpoint slices will be the preferred method, as it creates less
+indirection and gives explicit Kube API access to what IP addresses are being used by OVN-Kubernetes.
```

Comment:
I think you can add link to https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/docs/features/multiple-networks/mirrored-endpointslices.md to say "see this for more information"